src/components/fields/image-upload/image-review/image-review.tsx

```diff
@@ -0,0 +1,423 @@
+					.length < maxFiles)
+		) {
+			// image manager will handle the rest
+			setImages((prev) => {
+				return [
+					...prev,
+					{
+						file: selectedFile,
+						name: selectedFile.name,
+						dimensions,
+						status: EImageStatus.NONE,
+						uploadProgress: 0,
+						addedFrom: "reviewModal",
+						slot: ImageUploadHelper.findAvailableSlot(prev),
+					},
+				];
+			});
+		}
+	};
+
+	const handleDeleteDecision = (decision: boolean) => {
+		if (decision) {
+			setImages(images.filter((image, i) => i !== activeFileIndex));
+			setActiveFileIndex(Math.max(0, activeFileIndex - 1));
+		}
+		setActivePrompt(null);
+	};
+
+	const handleExitDecision = (decision: boolean) => {
+		if (decision) {
+			setImages(
+				images
+					.filter(
+						({ addedFrom, status }) =>
+							status >= EImageStatus.UPLOADED || (status < EImageStatus.NONE && addedFrom === "dragInput")
+					)
+					.map((file) => ({
+						...file,
+						drawingDataURL: undefined,
+						drawing: undefined,
+					}))
+			);
+			onExit();
+		}
+		setActivePrompt(null);
+	};
+
+	const handleSave = () => {
+		if (dispatchFieldEvent("save-review-images", id, { images, retry: handleSave })) {
+			setImages(
+				images
+					.filter(({ status }) => status >= EImageStatus.NONE)
+					.map((file) => {
+						const editedFile: IImage = { ...file };
+						if (file.drawingDataURL) {
+							editedFile.dataURL = file.drawingDataURL;
+						}
+						if (file.status < EImageStatus.UPLOAD_READY && file.status > EImageStatus.NONE) {
+							editedFile.status = EImageStatus.UPLOAD_READY;
+						}
+						editedFile.drawingDataURL = undefined;
+						editedFile.drawing = undefined;
+						return editedFile;
+					})
+			);
+			onExit();
+		}
+	};
+
+	// =============================================================================
+	// DRAWING
+	// =============================================================================
+	const handleStartDrawing = () => {
+		setDrawActive(true);
+		setActiveColor(PALETTE_COLORS[0].color);
+	};
+
+	const handleSaveDrawing = async () => {
+		const { dataURL: drawingDataURL, drawing } = imageEditorRef.current?.export() || {};
+		const image = await ImageHelper.dataUrlToImage(drawingDataURL);
+		const thumbnail = await ImageHelper.resampleImage(image, {
+			scale: 96 / image.width,
+			type: FileHelper.fileExtensionToMimeType(outputType),
+		});
+		const thumbnailDataURL = await FileHelper.fileToDataUrl(thumbnail);
+
+		setImages((prev) => {
+			const updatedFiles = [...prev];
+			updatedFiles[activeFileIndex] = {
+				...updatedFiles[activeFileIndex],
+				thumbnailDataURL,
+				drawingDataURL,
+				drawing,
+				status: compress ? EImageStatus.TO_RECOMPRESS : EImageStatus.EDITED,
+			};
+			return updatedFiles;
+		});
+
+		setDrawActive(false);
+		clearDrawingStates();
+	};
+
+	const handleClearDrawingDecision = (decision: boolean) => {
+		if (decision) {
+			setClearDrawing(true);
+			setTimeout(() => setClearDrawing(false));
```

Comment:
This timeout logic is not intuitive to me. Can the clear logic in image-editor be exposed as an imperative method on `useImperativeHandle`?

Reply:
added and also removed the `clear` prop because it's not intuitive indeed